kge/job/job.py

```diff
@@ -55,36 +58,92 @@ def __init__(self, config: Config, dataset: Dataset, parent_job: "Job" = None):
             for f in Job.job_created_hooks:
                 f(self)
 
-    def resume(self, checkpoint_file: str = None):
-        """Load job state from last or specified checkpoint.
-
-        Restores all relevant state to resume a previous job. To run the restored job,
-        use :func:`run`.
+    @staticmethod
+    def create(
+        config: Config, dataset: Optional[Dataset] = None, parent_job=None, model=None
+    ):
+        from kge.job import TrainingJob, EvaluationJob, SearchJob
 
-        Should set `resumed_from_job` to the job ID of the previous job.
+        if dataset is None:
+            dataset = Dataset.load(dataset)
+
+        job_type = config.get("job.type")
+        if job_type == "train":
+            return TrainingJob.create(
+                config, dataset, parent_job=parent_job, model=model
+            )
+        elif job_type == "search":
+            return SearchJob.create(config, dataset, parent_job=parent_job)
+        elif job_type == "eval":
+            return EvaluationJob.create(
+                config, dataset, parent_job=parent_job, model=model
+            )
+        else:
+            raise ValueError("unknown job type")
 
+    @classmethod
+    def create_from(
+        cls,
+        checkpoint: Optional[Union[str, Dict]] = None,
+        overwrite_config: Config = None,
+        dataset: Dataset = None,
+        parent_job=None,
+    ) -> Job:
         """
-        raise NotImplementedError
+        Creates a Job based on a checkpoint
+        Args:
+            checkpoint: path to checkpoint file or loaded checkpoint
```

Comment:
Why can't we just use a loaded checkpoint? I think we should have a utility method that takes a filename or a loaded checkpoint and returns the loaded checkpoint.

Reply:
just to make sure, before I restructure: we want to load the checkpoint before we call create_from?
then we would have to extend the method load_checkpoint to the signature load_checkpoint(checkpoint_file: Optional[str] = None, config: Optional[Config] = None). 
This method then loads the checkpoint_file, if provided or looks for the correct checkpoint, if only the config is provided.

Reply:
We can't we just pass a loaded checkpoint (or None) instead of Union[str,Dict]? Why does load_checkpoint require the config and not just the filename?

BTW: I think we should use clearer arguments/variable names throughout: `checkpoint` (loaded), `checkpoint_file` (filename), `checkpoint_or_file` (union) or so.

Reply:
checkpoint loading is now completely separated from job.create_from